### ISSUE TITLE
BFGS Segmentation Fault

### ISSUE THREAD

**Issue by moody-marlin** [dask/dask-glm#14]
On my local machine, `bfgs()` is creating a segmentation fault anytime I try to run it.

```
import dask.array as da
from dask_glm.utils import make_y
from dask_glm.logistic import bfgs

X = da.random.random((100,2), chunks=(20,2))
y = make_y(X)
bfgs(X, y)
```

yields:
```
//anaconda/lib/python3.5/site-packages/dask/array/core.py:464: RuntimeWarning: overflow encountered in true_divide
  o = func(*args, **kwargs)
Segmentation fault: 11
```

Possibly related to https://github.com/dask/dask/pull/1927 ?

**Comment by mrocklin** [comment]
The warning is likely unrelated.  

Can you try my dev branch?

**Comment by moody-marlin** [comment]
Your dev branch doesn't give me the segmentation fault - what's different?

**Comment by mrocklin** [comment]
Some things broke during the merge.  They're fixed now.

**Comment by moody-marlin** [comment]
Fixed with #10
